internal/driver/driver.go

```diff
@@ -463,14 +467,26 @@ func (d *Driver) RemoveDevice(deviceName string, protocols map[string]models.Pro
 func (d *Driver) RefreshExistingDevicePaths() {
 	d.lc.Debug("Refreshing existing device paths")
 	for _, cd := range d.ds.Devices() {
-		fdPath := cd.Protocols[UsbProtocol][Path].(string)
-		cn, sn, err := getUSBDeviceIdInfo(fdPath)
+		d.RefreshExistingDevicePath(cd)
+	}
+}
+
+// RefreshExistingDevicePaths checks whether the existing devices match the connected devices.
+// If there is a mismatch between them, scan all paths to find the matching device and update the existing device with the correct path.
+func (d *Driver) RefreshExistingDevicePath(cd models.Device) {
+	d.lc.Debug("Refreshing existing device paths")
+	for _, fdPath := range cd.Protocols[UsbProtocol][Paths].([]interface{}) {
```

Comment:
I am not sure if this is the proper approach. I think it might make more sense to find all of the /dev/video that this device uses and then compare them.